aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,348 @@
+| `Exception`                 | `Error`      |            | ✅                | ✅                    | |
+| `JSObject`                  | `Object`     |            | ✅                | ✅                    | ✅ |
+| `String`                    | `String`     |            | ✅                | ✅                    | ✅ |
+| `Object`                    | `Any`        |            | ✅                |                       | ✅ |
+| `Span<Byte>`                | `MemoryView` |            |                   |                        | |
+| `Span<Int32>`               | `MemoryView` |            |                   |                        | |
+| `Span<Double>`              | `MemoryView` |            |                   |                        | |
+| `ArraySegment<Byte>`        | `MemoryView` |            |                   |                        | |
+| `ArraySegment<Int32>`       | `MemoryView` |            |                   |                        | |
+| `ArraySegment<Double>`      | `MemoryView` |            |                   |                        | |
+| `Task`                      | `Promise`    |            |                   | ✅                    | |
+| `Action`                    | `Function`   |            |                   |                        | |
+| `Action<T1>`                | `Function`   |            |                   |                        | |
+| `Action<T1, T2>`            | `Function`   |            |                   |                        | |
+| `Action<T1, T2, T3>`        | `Function`   |            |                   |                        | |
+| `Func<TResult>`             | `Function`   |            |                   |                        | |
+| `Func<T1, TResult>`         | `Function`   |            |                   |                        | |
+| `Func<T1, T2, TResult>`     | `Function`   |            |                   |                        | |
+| `Func<T1, T2, T3, TResult>` | `Function`   |            |                   |                        | |
+
+The following conditions apply to type mapping and marshalled values:
+
+* The :::no-loc text="Array of"::: column indicates if the .NET type can be marshalled as a JS [`Array`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array). Example: C# `int[]` (`Int32`) mapped to JS `Array` of `Number`s.
+* When passing a JS value to C# with a value of the wrong type, the framework throws an exception in most cases. The framework doesn't perform compile-time type checking in JS.
+* `JSObject`, `Exception`, `Task` and `ArraySegment` create `GCHandle` and a proxy. You can trigger disposal in developer code or allow [.NET garbage collection (GC)](/dotnet/standard/garbage-collection/) to dispose of the objects later. These types carry significant performance overhead.
+* [`IMemoryView`](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/marshal.ts)
+  * `MemoryView` is a .NET type, not a native JS type.
```

Comment:
It's a bit weird to say `MemoryView` is a .NET type, when think we're actually talking about a JS data structure, correct?

Reply:
It is JavaScript "type" because they mostly don't have "types". It's a shape of a JS object which .NET consistently creates. I don't know how to say it nicely. "not a browser type." would be better

Reply:
I can't grok what the sentence should state from that discussion. Can you provide the full sentence that you'd like to see there ...

> `MemoryView` is a ...

Reply:
Sorry.

`IMemoryView` is a JS type of .NET interop, not a primitive JS type.